src/CMakeLists.txt

```diff
@@ -69,17 +69,38 @@ if ("${JUPYTER}" STREQUAL "ON")
    add_definitions(-DRB_XCODE)
 endif()
 
+#MESSAGE("Boost location hints:")
+#if (BOOST_ROOT)
+#  MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
+#else()
+#  MESSAGE("  BOOST_ROOT not set")
+#endif()
+#if (BOOST_INCLUDEDIR)
+#  MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+#endif()
+#if (BOOST_LIBRARYDIR)
+#  MESSAGE("  BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+#endif()
 MESSAGE("Boost location hints:")
+if (BOOST_ROOT OR BOOST_INCLUDEDIR OR BOOST_LIBRARYDIR) 
+  set(Boost_NO_SYSTEM_PATHS ON)
+  set(Boost_NO_BOOST_CMAKE ON)
+  set(Boost_NO_WARN_NEW_VERSION ON)
+endif()
 if (BOOST_ROOT)
   MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
-else()
-  MESSAGE("  BOOST_ROOT not set")
+  set(Boost_INCLUDE_DIR ${BOOST_ROOT}/include)
+  set(Boost_LIBRARY_DIR ${BOOST_ROOT}/lib)
+  INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 endif()
 if (BOOST_INCLUDEDIR)
   MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+  set(Boost_INCLUDE_DIR ${BOOST_ROOT})
+  INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 endif()
 if (BOOST_LIBRARYDIR)
   MESSAGE("  BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+  set(Boost_LIBRARY_DIR ${BOOST_LIBRARYDIR})
```

Comment:
Here, I just think that we should allow users to set (or not set) `-DBoost_LIBRARY_DIR` directly.

Reply:
We could try that, but I am not confident in the behavior of cmake when its special boost variables are not set. Maybe this would be fine when using with Boost_NO_SYSTEM_PATHS and Boost_NO_BOOST_CMAKE.